tools/restatectl/src/commands/log/dump_log.rs

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::path::PathBuf;
+use std::time::Duration;
+
+use anyhow::Context;
+use cling::prelude::*;
+
+use futures_util::StreamExt;
+use restate_bifrost::{BifrostService, FindTailAttributes};
+use restate_core::network::net_util::create_tonic_channel_from_advertised_address;
+use restate_core::network::{MessageRouterBuilder, Networking};
+use restate_core::{MetadataBuilder, MetadataManager, TaskCenter, TaskCenterBuilder, TaskKind};
+use restate_metadata_store::local::LocalMetadataStoreService;
+use restate_metadata_store::MetadataStoreClient;
+use restate_rocksdb::RocksDbManager;
+use restate_server::config_loader::ConfigLoaderBuilder;
+use restate_types::config::{
+    Configuration, MetadataStoreClientOptions, MetadataStoreOptions, RocksDbOptions,
+};
+use restate_types::live::{BoxedLiveLoad, Live};
+use restate_types::logs::{KeyFilter, LogId, Lsn, SequenceNumber};
+use restate_types::retries::RetryPolicy;
+use restate_wal_protocol::Envelope;
+use tonic_health::pb::health_client::HealthClient;
+use tonic_health::pb::HealthCheckRequest;
+use tracing::{debug, info, warn};
+
+use crate::build_info;
+
+#[derive(Run, Parser, Collect, Clone, Debug)]
+#[clap()]
+#[cling(run = "dump_log")]
+pub struct DumpLogOpts {
+    /// Set a configuration file to use for Restate.
+    /// For more details, check the documentation.
+    #[arg(
+        short,
+        long = "config-file",
+        env = "RESTATE_CONFIG",
+        value_name = "FILE"
+    )]
+    config_file: Option<PathBuf>,
+    /// Specifies the log_id to dump.
+    #[arg(short, long = "log")]
```

Comment:
Minor nit - let this be `--log-id` for consistency with `logs describe`:

```suggestion
    #[arg(short, long)]
```